packages/core/src/interpreter.ts

```diff
@@ -883,7 +888,8 @@ export class Interpreter<
             ? serviceCreator(context, _event.data, {
                 data: resolvedData,
                 src: invokeSource,
-                meta: activity.meta
+                meta: activity.meta,
+                self: toActorRef(this)
```

Comment:
Re chosen naming - I wonder how using `self` will play out here. It will get used in callbacks etc and I wonder if using `parent.send()` wouldn't make more sense in that context.

Reply:
Also - it would probably make sense to create this once, instead of recreating it for each invoke. This ensures that those refs are isolated between invokes but I don't see why this would matter in practice.

Reply:
`self` is what is used in Akka, and it makes sense to me: it is a _self-reference_. Naming it something like `parent` would get confusing, and actually be completely wrong, since it's not a reference to the parent of this actor.

The alternative is calling it `ref`, `selfRef`, etc. I'd prefer to stick with Akka naming.

Reply:
> Also - it would probably make sense to create this once, instead of recreating it for each invoke. This ensures that those refs are isolated between invokes but I don't see why this would matter in practice.

This is done in 2a6a822f89a6c120b586f0cc17de281df5e88d67

Reply:
While the points related to `.origin` raised in https://github.com/statelyai/xstate/pull/2887#issue-1086967204 were invalid, I think that we should still revisit naming this "self". In Akka (maybe just in Classic, but that doesn't affect naming considerations) one gets access to both `context.self` and `context.parent` and I think that was is provided here is much closer to `context.parent`.

It's because it's the child actor that sends an event to its parent and not either that parent sending an event to self or that child sending an event to self.

Reply:
Maybe `invoker`?

Reply:
`invoker` would work OK. Any particular objections to just `parent` though?

Reply:
> `invoker` would work OK. Any particular objections to just `parent` though?

Just worried it would be confused with "parent of this machine" or "parent of the invoked actor"

Reply:
Can't something like "invoker" be misinterpreted in the same way? Both names are sensitive to the context and if somebody assumes a different context here then they would misinterpret both in the same way. I guess maybe `invoker` is harder to misinterpret but it's also not that clear if it's the invoker of this machine of the invoker of the invoked actor 🤔 

I've gone back to other comments by @ivandotv and @erikras though and I think that after all the chosen name (self) is right. I still believe that this might be somewhat confusing because very often u might want to `self.send()` from within a child actor and this won't really send to its own "self" but this is somewhat a closure/nesting problem. At the very time when `serviceCreator` gets called, there is no "self" for the child actor yet - and if we'd make this available to other types of implementations (such as actions) then there would be no "self" for those implementation types at all. So whatever is given at this very position/time in the `meta` param has to refer to the machine that is "calling" those implementations and `self` suits that.

<img src="https://media.giphy.com/media/6GpKLRN7FcKztT8fG1/giphy.gif" />

Reply:
From the Akka docs:

> **parent**
> There is no `parent` in Typed. Instead you have to explicitly include the `ActorRef` of the parent as a parameter when constructing the Behavior.
>
> The reason for not having a parent in Typed is that it wouldn’t be possible to know the type for the parent ActorRef[T] at compile time without having an additional type parameter in the Behavior. For testing purposes it’s also better to pass in the parent since it can be replaced by a probe or being stubbed out in tests.

Reply:
We can also provide good patterns in the docs for naming:

```js
invoke:

Reply:
> The reason for not having a parent in Typed is that it wouldn’t be possible to know the type for the parent ActorRef[T] at compile time without having an additional type parameter in the Behavior. For testing purposes it’s also better to pass in the parent since it can be replaced by a probe or being stubbed out in tests.

Yeah, I agree with this reasoning - if we follow that advice though then we should deprecate `sendParent` though (and similar things too, like `origin`) and the question is: is it pragmatic for us to deprecate it? 😅

Reply:
This `self` thing can be a little bit confusing but the example that David has provided is pretty self-explanatory.

Reply:
> self-explanatory

I see what you did there @ivandotv 😂